service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ManageLegalCounselControllerSubmittedTest.java

```diff
@@ -126,14 +121,14 @@ private CaseDetails buildCaseDetailsWithGivenChildren(List<Element<Child>> child
             .build();
         return CaseDetails.builder()
             .id(TEST_CASE_ID_AS_LONG)
-            .data(caseConverter.toMap(previousCaseData))
+            .data(toMap(previousCaseData))
```

Comment:
vv minor but not entirely what I meant, there is a method in one of the base classes that is called as case details, which will do the conversion for you. You can also set the id in the case data and it will know to pull it out and add it to the id in the case details, the case converter does this part

Reply:
oh ok. will have a look.

Reply:
I couldn't find this method. If you talking about the "postSubmittedEvent" method, I think this won't work as I need both the previous and the current case data objects.
Or did I misunderstand you?

Reply:
So what I think I meant was that you can do `caseDataBuilder.id(...)` and then there is a method called `toCallbackResponse(caseData, caseDataBefore)` which can be called and then passed to the `postSubmittedEvent` method

Reply:
I can't find this method.
Shall we have a call?

Reply:
About this one: I'm now using the method you're saying, but I couldn't pass the CaseData object.
Because the list of children is pointing (by reference) to the same child objects, I have to map the CaseData to a Map before making changes to the Child objects.
So, I'm using the method you suggested, but I had to pass the CaseDetails as parameters instead.